### PR TITLE
chore(cognito): reorganize identity providers structure so that UserPoolIdentityProviderBase is not exported

### PR DESCRIPTION
**Pull request by shivlaks**
we had split our identity providers to have a file per provider.
however, in this organization we were exporting the base class which
we did not intend to.

the `UserPoolIdentityProviderBase` class cannot be moved to a separate
file (i.e. private folder) as JSII will be unable to resolve type.

updated to follow our typical pattern of containing all extensions of the base
class within a single file `user-pool-idp.ts` and moved all the tests
over as well.

BREAKING CHANGE: the `UserPoolIdentityProviderBase` abstract class has been removed. Use the `UserPoolIdentityProvider*` classes directly.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts [hunk 1]
```diff
@@ -29,4 +32,388 @@ export class UserPoolIdentityProvider {
...
+   * Specify custom attribute mapping here and mapping for any standard attributes not supported yet.
+   * @default - no custom attribute mapping
+   */
+  readonly custom?: { [key: string]: ProviderAttribute };
+}
+
+/**
+ * Properties to create a new instance of UserPoolIdentityProvider
+ */
+export interface UserPoolIdentityProviderProps {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts:220]: Can this also be non-exported?

**Review by nija-at — CHANGES_REQUESTED** [review]
How about instead moving UserPoolIdentifyProviderBase off into a separate file, maybe under `lib/private/`? How complicated would that look.
I like the separate file structure we already have and want to see if we can maintain that.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts (hunk 1 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts:220]: > How about instead moving UserPoolIdentifyProviderBase off into a separate file, maybe under `lib/private/`? How complicated would that look?
> I like the separate file structure we already have and want to see if we can maintain that.

this is actually what I initially tried (mentioned it in the commit body) but JSII will not allow it. I also tried to preserve the structure initially.

as for the props,  we need to export it as it's an extended interface. All our constructs export the base interface. (jsii also throws an error if you exclude it)

Comment (nija-at) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts:220]: Ok, that's fine.

Can we open issues in JSII for these two? I'd like to hear from the JSII team if this is unsupported for a reason or just a gap.

**Review by nija-at — APPROVED** [review]

File: packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts (hunk 1 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-idp.ts:220]: will do!

**Pull request merged**